pkg/kdump/kdump-client.es6

```diff
@@ -0,0 +1,183 @@
+ *      config:    settings from kdump.conf
+ *      target:    dump target info, content depends on dump type
+ *                 always contains the keys:
+ *                     target          value in ["local", "nfs", "ssh", "raw", "mount", "unknown"]
+ *                     multipleTargets true if the config file has more than one target defined, false otherwise
+ *  }
+ *
+ */
+export class KdumpClient {
+    constructor() {
+        var self = this;
+        self.state = {
+            installed: undefined,
+            enabled: undefined,
+            state: undefined,
+            config: undefined,
+            target: undefined,
+        };
+        cockpit.event_target(self);
+
+        // listen to service status changes
+        self.kdumpService = serviceProxy("kdump");
+        self.kdumpService.addEventListener('changed', function() {
+            self.state.installed = self.kdumpService.exists;
+            self.state.enabled = self.kdumpService.enabled;
+            self.state.state = self.kdumpService.state;
+            self.dispatchEvent("kdumpStatusChanged", self.state);
+        });
+
+        // watch the config file
+        self.configClient = new ConfigFile("/etc/kdump.conf", true);
+        // catch config changes
+        self.configClient.addEventListener('kdumpConfigChanged', function() {
+            self.state.config = self.configClient.settings;
+            self.state.target = self.targetFromSettings(self.configClient.settings);
+            self.dispatchEvent("kdumpStatusChanged", self.state);
+        });
+    }
+    ensureOn() {
+        // we consider the state to be "on" when it's enabled and running
+        return cockpit.all(
+            this.kdumpService.enable(),
+            this.kdumpService.start()
+        );
+    }
+    ensureOff() {
+        // we consider the state to be "off" when it's disabled and stopped
+        return cockpit.all(
+            this.kdumpService.stop(),
+            this.kdumpService.disable()
+        );
+    }
+    crashKernel() {
+        // crash the system kernel
+        return cockpit.spawn(["/bin/sh", "-c", "echo 1 > /proc/sys/kernel/sysrq && echo c > /proc/sysrq-trigger"], { superuser: "require" });
+    }
+    testWriteLocation(settings) {
+        var target = this.targetFromSettings(settings);
+        var path;
+        if (target && target.path)
+            path = target.path;
+        // if path is invalid or we haven't set one, use default
+        if (!path)
+            path = "/var/crash";
+
+        var dfd = cockpit.defer();
+        var filename = path;
+        if (!target || target.target != "local")
+            dfd.resolve();
+        else {
+            // local path, try to see if we can write
+            if (filename.slice(-1) != "/")
+                filename += "/";
+            filename += "cockpit.write.tmp";
+            cockpit.spawn(["/bin/sh", "-c", "touch " + filename + " && rm " + filename], { superuser: "try" } )
```

Comment:
Which directory does this happen in? Should it just happen in the home directory? Likely /var/crash or /tmp or $TEMP is better, no?